controller: expose reconciliation state and progress

pritesh-pagar opened the pull request:
- Add reconciliation phase and progress in a status field
- Expose reconciliation status as a prom metric

pritesh-pagar committed:
Add reconciliation phase in status field

* Status will have a new field Phase to show reconciliation phase, such
  as - Ready, ApplyingChanges, Invalid

pritesh-pagar committed:
Merge pull request #4 from pritesh-pagar/status-branch

Add reconciliation phase in status field

pritesh-pagar committed:
controllers: expose controller phase as metric

pritesh-pagar committed:
controller: address review comments to expose phase as metric

pritesh-pagar committed:
controllers: expose controller phase as metric

controllers: expose controller phase as metric

pritesh-pagar committed:
controller: show reconciliation progress in status

- add a new field reconcile progress in status do donote what exactly
  reconciler is doing
- expose progress as a prom metrics as well

pritesh-pagar committed:
Merge pull request #6 from pritesh-pagar/status-branch

controller: show reconciliation progress in status

timoha reviewed (approved):
LG, thanks for the contribution!

tomasbanet committed:
Add zkroot flag

Allow overriding the default hbase znode parent used by gohbase
admin client

Needed for running multiple hbase clusters on the same zookeeper
cluster, e.g. for replication between hbases.

Tested by running `go run ./cmd/ -zkroot=/hbase-test` and observing

```
failed to read the /hbase-test/master znode: zk:
could not connect to a server
```

in the output.

tomasbanet committed:
Upgrade gohbase (#8)

* Upgrade gohbase

```
go get github.com/tsuna/gohbase
go mod tidy
```

* Migrate from github.com/golang/mock to https://github.com/uber-go/mock
